src/openzaak/utils/tasks.py

```diff
@@ -0,0 +1,765 @@
+
+    for field in gegevensgroep_fields:
+        gegevens_groep_value = data.pop(field, {})
+
+        for key, value in gegevens_groep_value.items():
+            data[f"{field}_{key}"] = value
+
+    data["uuid"] = str(uuid4())
+
+    if "vertrouwelijkheidaanduiding" not in data:
+        informatieobjecttype = data["informatieobjecttype"]
+        data[
+            "vertrouwelijkheidaanduiding"
+        ] = informatieobjecttype.vertrouwelijkheidaanduiding
+
+    instance = EnkelvoudigInformatieObject(**data)
+
+    if not instance.identificatie:
+        instance.identificatie = generate_unique_identification(
+            instance, "creatiedatum"
+        )
+
+    try:
+        instance.clean()
+    except ValidationError as e:
+        error_message = (
+            "A validation error occurred while validating a "
+            "EnkelvoudigInformtatieObject on line %(row_index)s: \n"
+            "%(error)s"
+        ) % dict(row_index=row_index, error=str(e))
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    file_path = document_row.bestandspad
+    path = Path(file_path)
+
+    if not path.exists() or not path.is_file():
+        error_message = (
+            "The given filepath %(file_path)s does not exist or is a file for "
+            "row %(row_index)s"
+        ) % dict(file_path=file_path, row_index=row_index)
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    default_dir = _get_default_path(instance.inhoud)
+    import_path = f"{default_dir}/{path.name}"
+
+    if not default_dir.exists():
+        default_dir.mkdir(parents=True)
+
+    try:
+        shutil.copy2(file_path, import_path)
+    except Exception as e:
+        error_message = (
+            "Unable to copy file for row %(row_index)s: \n %(error)s"
+        ) % dict(error=str(e), row_index=row_index)
+
+        logger.warning(error_message)
+        document_row.comment = error_message
+        document_row.processed = True
+
+        return document_row
+
+    instance.inhoud.name = str(import_path)
+
+    document_row.instance = instance
+    return document_row
+
+
+@transaction.atomic()
+def _batch_create_eios(batch: list[DocumentRow]) -> None:
+    canonicals = []
+
+    for row in batch:
+        if row.instance is None:
+            continue
+
+        canonical = EnkelvoudigInformatieObjectCanonical()
+        row.instance.canonical = canonical
+
+        canonicals.append(canonical)
+
+    try:
+        EnkelvoudigInformatieObjectCanonical.objects.bulk_create(canonicals)
+    except IntegrityError as e:
+        for row in batch:
+            row.processed = True
+            row.comment = f"Unable to load row due to batch error: {str(e)}"
+
+        raise e
+
+    try:
+        eios = EnkelvoudigInformatieObject.objects.bulk_create(
```

Comment:
There is should be a permission check before creating a document

I also have a question - should we check that the content of the document is reachable?

Reply:
> I also have a question - should we check that the content of the document is reachable?

I think this is done as part of `_import_document_row`? After checking if the path exists and is a file, it's copied (and if the new destination would be unreachable, I assume the copy operation would fail)

Reply:
Regarding checking the content of the document, this is checked [here](https://github.com/open-zaak/open-zaak/pull/1654/files#diff-57eb99f9dc886fbdf3d28e3a5b45c37d37566f11957dc93682fc13fe71122b4aR433).